source/client-side-encryption/client-side-encryption.md

```diff
@@ -0,0 +1,2483 @@
+`$CreateEncryptedCollection(CE, database, collName, collOpts, kmsProvider, masterKey)$` is defined, where `$CE$` is a
+[ClientEncryption](#clientencryption-1) object, `$kmsProvider$` is a [KMSProvider](#KMSProvider) and `$masterKey$` is
+equivalent to the `$masterKey$` defined in [DataKeyOpts](#datakeyopts). It has the following behavior:
+
+- If `$collOpts$` contains an `"encryptedFields"` property, then `$EF$` is the value of that property. Otherwise, report
+  an error that there are no `encryptedFields` defined for the collection.
+- Let `$EF'$` be a copy of `$EF$`. Update `$EF'$` in the following manner:
+  - Let `$Fields$` be the `"fields"` element within `$EF'$`.
+  - If `$Fields$` is present and is an array value, then for each element `$F$` of `Fields`:
+    - If `$F$` is not a document element, skip it.
+    - Otherwise, if `$F$` has a `"keyId"` named element `$K$` and `$K$` is a `null` value:
+      - Create a [DataKeyOpts](#datakeyopts) named `$dkOpts$` with the `$masterKey$` argument.
+      - Let `$D$` be the result of `CE.createDataKey(kmsProvider, dkOpts)`.
+      - If generating `$D$` resulted in an error `$E$`, the entire `$CreateEncryptedCollection$` must now fail with
+        error `$E$`. Return the partially-formed `$EF'$` with the error so that the caller may know what datakeys have
+        already been created by the helper.
+      - Replace `$K$` in `$F$` with `$D$`.
+- Create a new set of options `$collOpts'$` duplicating `$collOpts$`. Set the `"encryptedFields"` named element of
+  `$collOpts'$` to `$EF'$`.
+- Invoke the `CreateCollection` helper as `$CreateCollection(database, collName, collOpts')$`. Return the resulting
+  collection and the generated `$EF'$`. If an error occurred, return the resulting `$EF$` with the error so that the
+  caller may know what datakeys have already been created by the helper.
+
+Drivers MUST document that `$createEncryptedCollection$` does not affect any auto encryption settings on existing
+MongoClients that are already configured with auto encryption. Users must configure auto encryption after creating the
+encrypted collection with the `$createEncryptedCollection$` helper.
+
+#### Drop Collection Helper
+
+Drivers MUST support a BSON document option named `encryptedFields` for any
+[drop](https://www.mongodb.com/docs/manual/reference/command/drop) command helpers (e.g. `Database.dropCollection()`,
+`Collection.drop()`). This option will only be interpreted by the helper method and MUST NOT be passed to the
+[drop](https://www.mongodb.com/docs/manual/reference/command/drop) command.
+
+> \[!NOTE\]
+>
+> Users are not expected to set the `escCollection` and `ecocCollection` in `encryptedFields`. SERVER-74069 added
+> server-side validation for those fields and no longer allows names to deviate from the following:
+>
+> - `enxcol_.<collectionName>.esc`
+> - `enxcol_.<collectionName>.ecoc`
+>
+> Drivers SHOULD NOT document the `escCollection` and `ecocCollection` options.
```

Comment:
@kevinAlbs: Unrelated to the conversion but should this be "MUST NOT" to be consistent with the create note above?

```suggestion
> Drivers MUST NOT document the `escCollection` and `ecocCollection` options.
```